From b5a8f0cd8b779843fdf961263b1eeed9895f2b04 Mon Sep 17 00:00:00 2001
From: effects3d <effects3dee@gmail.com>
Date: Sun, 13 Sep 2026 14:59:49 +0200
Subject: [PATCH 10/11] Partially revert "fixup! loader: HACK: Build wine64 and
 wine64-preloader."

Fixes wineboot from crashing at launch.

Source: Proton-Wine commit: 6ee0f43b577d283e6dc539940ab972e642f1f4ec
Signed-off-by: effects3d <effects3dee@gmail.com>
---
 dlls/ntdll/unix/loader.c | 11 ++---------
 1 file changed, 2 insertions(+), 9 deletions(-)

diff --git a/dlls/ntdll/unix/loader.c b/dlls/ntdll/unix/loader.c
index 8d3a110e55c..ac44380583a 100644
--- a/dlls/ntdll/unix/loader.c
+++ b/dlls/ntdll/unix/loader.c
@@ -549,15 +549,8 @@ static void preloader_exec( char **argv, WORD machine )
 {
 #ifdef HAVE_WINE_PRELOADER
 #if !defined(__arm__) && !defined(__aarch64__)
-    const char *p;
-
-    if (machine == IMAGE_FILE_MACHINE_AMD64
-        && ((p = remove_tail( argv[1], "x86_64-unix/wine" )) || (p = remove_tail( argv[1], "i386-unix/wine64" ))))
-    {
-        asprintf( &argv[0], "%si386-unix/wine64-preloader", p );
-        asprintf( &argv[1], "%si386-unix/wine64", p );
-    }
-    else
+    if (machine == IMAGE_FILE_MACHINE_AMD64)
+        asprintf( &argv[0], "%s64-preloader", argv[1] );
 #endif
         asprintf( &argv[0], "%s-preloader", argv[1] );
 
-- 
2.55.0

